Source/Core/Core/NetPlayServer.h

```diff
@@ -71,6 +73,8 @@ class NetPlayServer : public TraversalClientClient
     u32 ping;
     u32 current_game;
 
+    NetPlayQoSSession* qos_session;
```

Comment:
This should be a `std::unique_ptr`

Reply:
` error: no matching function for call to ‘std::pair<const unsigned char, NetPlayServer::Client>::pair(unsigned char&, NetPlayServer::Client&)’`

I guess using a `std::unique_ptr` in the `Client` class makes it not a POD type?

Reply:
This is likely because the

```cpp
m_players.emplace(*(PlayerId*)player.socket->data, player);
``` 

call in `OnConnect()` needs `to std::move()` `player`.

The

```cpp
if (SConfig::GetInstance().bQoSEnabled)
  player.qos_session = new NetPlayQoSSession(player.socket);
```

condition following the `emplace` would need to be moved before it as well (otherwise it's accessing a moved from object).

Reply:
`std::move` works. Fixed.